apis/v1alpha1/opampbridge_types.go

```diff
@@ -0,0 +1,144 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OpAMPBridgeSpec defines the desired state of OpAMPBridge.
+type OpAMPBridgeSpec struct {
+	// OpAMP backend Server endpoint
+	// +required
+	Endpoint string `json:"endpoint"`
+	// Protocol scheme for the communication between OpAMP server and OpAMP Bridge
+	// +required
+	Protocol string `json:"protocol"`
+	// Capabilities supported by the OpAMP Bridge
+	// +required
+	Capabilities []OpAMPBridgeCapability `json:"capabilities"`
+	// ComponentsAllowed is a list of allowed OpenTelemetry components for each pipeline type (receiver, processor, etc.)
+	// +optional
+	ComponentsAllowed map[string][]string `json:"components_allowed,omitempty"`
```

Comment:
Could we adjust this json field to be camel case? `componentsAllowed` to match the rest of our styles?

Reply:
Yes, it should be camel case. But, in Opamp bridge application code base, the yaml tag used is `components_allowed`, so will it create a confusion ? If so, then should we change that as well ?

https://github.com/open-telemetry/opentelemetry-operator/blob/423ff4c52320921e38e3856717383c4673479210/cmd/operator-opamp-bridge/config/config.go#L42-L49

Reply:
Yes, we should change that too... I can make a PR today for that.

Reply:
it limits the bridge's capabilities to only create collectors with some set of components to prevent someone from creating a collector with unknown exporters for example.